### PR TITLE
fix: Change minimum number of teams required for org self-serve

### LINKED ISSUE #15812: [CAL-4051] Error with the UI teams
UI teams tab showing only one team but says "you have a lot of teams".

profile with only one team.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4051](https://linear.app/calcom/issue/CAL-4051/erorr-with-the-ui-teams)</sub>

### PR DESCRIPTION
**Pull request by Souptik2001**
## What does this PR do?

Change minimum number of teams required for org self-serve

- Fixes #15812
- Fixes CAL-4051

https://www.loom.com/share/12c9197281164a79997ae17756a93528?sid=d63edb2e-1d07-434f-9f46-6a9accdc62e5

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Please check the video walkthrough above.

### CODE REVIEW

File: packages/lib/constants.ts [hunk 1]
```diff
@@ -152,7 +152,8 @@ export const BOOKER_NUMBER_OF_DAYS_TO_LOAD = parseInt(
 export const CLOUDFLARE_SITE_ID = process.env.NEXT_PUBLIC_CLOUDFLARE_SITEKEY;
 export const MINIMUM_NUMBER_OF_ORG_SEATS = 30;
 export const ORG_SELF_SERVE_ENABLED = process.env.NEXT_PUBLIC_ORG_SELF_SERVE_ENABLED === "1";
-export const ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE = 0;
+export const ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE =
+  process.env.NEXT_PUBLIC_ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE || 2;
```
Comment (Souptik2001) [inline packages/lib/constants.ts:156]: Introduced a new env variable here, just to give flexibility to change this value, but the default value is 2, i.e someone needs to create **two or more** teams **to view that option** and **also to be able to create a org**.

Comment (anikdhabal) [inline packages/lib/constants.ts:156]: I don't see any reason to create a new environment here; we can achieve a similar result by setting a value directly for `ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE`.

I think the main reason for the issue is setting `ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE` to zero

**Comment by anikdhabal** [comment]
@Souptik2001 left a suggestion

File: packages/lib/constants.ts (hunk 1 above)
Comment (Souptik2001) [reply packages/lib/constants.ts:156]: @anikdhabal Ok if you are saying we don't need the env var, then let me remove it.

Anyhow the default value was 2 only. I did add the env var, just for flexibility. For example if someone self-hosts it and wanted to set it to 3 instead, then they can set it using that env var. 👍 

I am frankly not sure if that requirement will arise or not, but still just confirming once again before removing the env var, that should I remove it, whatever you suggest? 🙌

Comment (anikdhabal) [reply packages/lib/constants.ts:156]: Okay keep it. Yeah, it would be beneficial for self-host. Pls add it in .env.example also

**Review by anikdhabal — DISMISSED** [review]
LGTM!

**Review by keithwillcode — CHANGES_REQUESTED** [review]
https://github.com/calcom/cal.com/commit/b310eb1a7eab03013fe76cbff85d61cdfcd1a1c9

This was intentionally removed recently. 

Why are we changing it back to 2?

**Comment by anikdhabal** [comment]
> [b310eb1](https://github.com/calcom/cal.com/commit/b310eb1a7eab03013fe76cbff85d61cdfcd1a1c9)
> 
> This was intentionally removed recently.
> 
> Why are we changing it back to 2?

Hey keith, UI teams tab showing only one team but says "you have a lot of teams". This is the issue visible to user when it is zero.

**Comment by Souptik2001** [comment]
@anikdhabal @keithwillcode 

FYI - This does affect the minumum number of teams required to be able to create an org. 

https://github.com/calcom/cal.com/blob/8508a6f1a083f8c01881760fcdba0f938ea87633/packages/trpc/server/routers/viewer/organizations/create.handler.ts#L92-L94

So, maybe that's why it was changed to 0, because the target might have been to allow the users to create org even when they had no teams.

So, if we want to allow them to be able to creat org even when they had no team, but we want to fix the UI also (which as per me also needs to be fixed), then we can introduce a new constant named `ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE_HELPER_TEXT` and make that value as 2 and keep `ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE` value as 0 as it is.

And use the `ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE_HELPER_TEXT` for the UI part.

What do you folks think?

**Comment by keithwillcode** [comment]
Yeah exactly. We removed the restriction of needing 2 teams to create an org. 

That should not be changed back just because a UI message is incorrect. 

Sounds like the proposal to use a different const could be a path forward.

**Comment by Souptik2001** [comment]
Perfect thanks @keithwillcode !

Updated to use a new const now. ✅ 

cc: @anikdhabal

File: packages/lib/constants.ts [hunk 2]
```diff
@@ -153,6 +153,8 @@ export const CLOUDFLARE_SITE_ID = process.env.NEXT_PUBLIC_CLOUDFLARE_SITEKEY;
 export const MINIMUM_NUMBER_OF_ORG_SEATS = 30;
 export const ORG_SELF_SERVE_ENABLED = process.env.NEXT_PUBLIC_ORG_SELF_SERVE_ENABLED === "1";
 export const ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE = 0;
+export const ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE_HELPER_DIALOGUE =
+  process.env.NEXT_PUBLIC_ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE_HELPER_DIALOGUE || 2;
```
Comment (keithwillcode) [inline packages/lib/constants.ts:157]: Nos that we have this separate value, I don’t think we need yet another env var for something so trivial.

Comment (keithwillcode) [inline packages/lib/constants.ts:157]:
```suggestion
2;
```

File: turbo.json [hunk 3]
```diff
@@ -415,6 +415,7 @@
     "NEXT_PUBLIC_HEAD_SCRIPTS",
     "NEXT_PUBLIC_BODY_SCRIPTS",
     "NEXT_PUBLIC_ORG_SELF_SERVE_ENABLED",
+    "NEXT_PUBLIC_ORG_MINIMUM_PUBLISHED_TEAMS_SELF_SERVE_HELPER_DIALOGUE",
```
Comment (keithwillcode) [inline turbo.json:418]:
```suggestion
```

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**